linkerd/app/outbound/src/discover/tests.rs

```diff
@@ -35,13 +35,19 @@ async fn errors_propagate() {
         })
     };
 
-    let profiles = support::profile::resolver().profile(addr, profiles::Profile::default());
+    let discover = {
+        let profiles = support::profile::resolver().profile(addr, profiles::Profile::default());
+        svc::mk(move |OrigDstAddr(addr)| {
+            // TODO(eliza): policy!
+            profiles.clone().oneshot(profiles::LookupAddr(addr.into()))
```

Comment:
does this test care about profiles or policies at all, really? What are we testing?

Reply:
i don't think this test actually cares besides that it needs something it can construct an inner stack from. if we remove the`From<(Option<profile::Receiver>, T)>` impl for `Discovery<T>` (making both a policy rx mandatory), we will need to return both a policy and profile rx here as well.

i believe this test _is_ testing the behavior of the `push_discover` stack, so we can't just remove that entirely.